### PR TITLE
util: make deepcopy in set_filtered_fields

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/sut_orchestrator/azure/platform_.py [hunk 1]
```diff
@@ -1170,9 +1170,11 @@ def _create_deployment_parameters(
             "availability_zones",
             "availability_type",
         ]
-        set_filtered_fields(self._azure_runbook, arm_parameters, copied_fields)
+        # the runbook may be updated, so make a copy to avoid the original runbook
+        azure_runbook = copy.deepcopy(self._azure_runbook)
+        set_filtered_fields(azure_runbook, arm_parameters, copied_fields)
         set_filtered_fields(
-            self._azure_runbook,
+            azure_runbook,
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: I don’t see the new run book is used later. Is it needed to set back?

Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/platform_.py]: this place will change the origin runbook. https://github.com/microsoft/lisa/blob/main/lisa/sut_orchestrator/azure/features.py#L2448

Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/platform_.py]: using deepcopy in set_filtered_fields also works

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py]: Change in the set_filtered_fields is right, but it needs to check regressions carefully.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**